cron_scripts/inactive_idle_accounts.pl

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+use Getopt::Long;
+use Data::Dumper;
+use JSON;
+use File::Slurp;
+use POSIX qw(strftime);
+use Pod::Usage qw(pod2usage);
+
+=head1 NAME
+
+inactive_idle_accounts.pl - to make idle accounts to inactive status
+
+=head1 SYNOPSIS
+
+  inactive_idle_accounts.pl [-help -debug] -last_login_days <#no. of days> -new_status <#zimbraAccountStatus>
+
+=head1 DESCRIPTION
+
+This script will go through Zimbra's Internal LDAP and look for any account that has a last login greater than the specified days, 
+the script will change their account status to new status that passed to the script
+
+=head1 OPTIONS
+
+=over
+
+=item -help
+
+Prints the help message.
+
+=item -last_login_days
+
+Specifies number of days to check for last login.
+
+=item -new_status
+
+Specifies new status that should be changed to target users.
+
+=item -debug
+
+Enable debug output.
+
+=back
+
+=cut
+
+my $dir_path;
+
+BEGIN {
+    use Cwd 'realpath';
+    ($dir_path) = __FILE__ =~ m{^(.*)/};    # $dir = path of current file
+    $dir_path = realpath($dir_path);
+}
+use lib "$dir_path/lib";
+use ZCS::CustomAPI;
+my $config_file = "$dir_path/config/config.json";
+
+my %opts = (
+    last_login_days => 97,
+    new_status      => 'locked'
+);
+
+GetOptions( \%opts, "help", "debug", "last_login_days=i", "new_status=s" )
+  or pod2usage( -verbose => 99, -sections => "SYNOPSIS", -exitval => 1 );
+
+pod2usage(1) if ( $opts{help} );
+
+my $config = JSON->new->utf8->relaxed->decode( scalar read_file $config_file);
+unless ( defined( $config->{MAILSTOREHOST} ) && $config->{MAILSTOREHOST} ne "" )
```

Comment:
Note: assuming you check $config as suggested you can simplify this to something like:
```
unless ( $config->{MAILSTOREHOST} )
```
Same goes for similar checks below